### PR TITLE
Implement /proc/net/tcp6

### PR DESCRIPTION
**Pull request by tanjianfeng**
Fixes: #829

Signed-off-by: Jianfeng Tan <henry.tjf@antfin.com>
Signed-off-by: Jielong Zhou <jielong.zjl@antfin.com>

### CODE REVIEW

**Comment by avagin** [comment]
Could you add a test case in test/syscalls/linux/proc_net_tcp.cc?

File: pkg/sentry/fs/proc/net.go [hunk 1]
```diff
@@ -500,7 +543,7 @@ func (n *netUDP) ReadSeqFileData(ctx context.Context, h seqfile.SeqHandle) ([]se
 	}
 
 	var buf bytes.Buffer
-	for _, se := range n.k.ListSockets() {
+	for _, se := range k.ListSockets() {
```
Comment (avagin) [inline pkg/sentry/fs/proc/net.go]: /b/f/w/pkg/sentry/fs/proc/net.go:546:21: undefined: k

File: pkg/sentry/fs/proc/net.go [hunk 2]
```diff
@@ -310,44 +310,51 @@ func networkToHost16(n uint16) uint16 {
...
+		addr1 := usermem.ByteOrder.Uint32(a.Addr[4:8])
+		addr2 := usermem.ByteOrder.Uint32(a.Addr[8:12])
+		addr3 := usermem.ByteOrder.Uint32(a.Addr[12:16])
+		fmt.Fprintf(w, "%08X%08X%08X%08X:%04X ", addr0, addr1, addr2, addr3, port)
+	}
 }
 
-// ReadSeqFileData implements seqfile.SeqSource.ReadSeqFileData.
-func (n *netTCP) ReadSeqFileData(ctx context.Context, h seqfile.SeqHandle) ([]seqfile.SeqData, int64) {
+func commonReadSeqFileDataTCP(n seqfile.SeqHandle, k *kernel.Kernel, ctx context.Context, h seqfile.SeqHandle, fa int, header []byte) ([]seqfile.SeqData, int64) {
```
Comment (avagin) [inline pkg/sentry/fs/proc/net.go]: /b/f/w/pkg/sentry/fs/proc/net.go:368:22: n.k undefined (type seqfile.SeqHandle is interface with no methods)

**Comment by tanjianfeng** [comment]
> Could you add a test case in test/syscalls/linux/proc_net_tcp.cc?

Done.

File: pkg/sentry/fs/proc/net.go (hunk 1 above)
Comment (tanjianfeng) [reply pkg/sentry/fs/proc/net.go]: Sorry about the compile error. Was introduced in code rebase. I shall have checked that before sending it out.

**Pull request merged**